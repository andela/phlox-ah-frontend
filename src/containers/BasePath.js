export * from '../requests/LoginRequests';
export * from '../util/AsyncUtil';
export * from '../actionTypes/UserConstants';
export * from '../actions/MsgInfoActions';
export * from '../components/Input/Input';
export * from '../components/Button/Button';
export * from '../requests/SignupRequests';
export * from '../requests/ArticleRequests';
export * from '../requests/ProfileRequest';

