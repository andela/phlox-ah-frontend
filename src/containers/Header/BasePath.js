export * from '../../actionTypes/UserConstants';
export * from '../../util/AsyncUtil';
