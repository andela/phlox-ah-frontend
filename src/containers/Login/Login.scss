#login-modal, #reset-password-modal, #forgot-password-modal{
  width:360px;
  input{
    margin: 0;
  }
  .close-modal {
    position: absolute;
    top: 8px;
    right: 10px;
  }
  .modal-footer{
    display:none;
  }
  .fa-facebook{
    color:#4267B2;
  }
  .fa-google-plus-square{
    color:#CD4539;;
  }
  .fa-twitter-square{
    color:#00ACED;;
  }
  .theme-color{
    color:#06576D;
  }
  #login-button, #forgot-button, #reset-password{
    width:100%;
    border-radius: 25px;
    background-color: #06576D;
    margin-top: 20px;
    font-weight: 600;
    height: 40px;

    i.fa-sign-in-alt {
      margin-left: 5px;
    }
  }
  h5, h6{
    font-weight: bold;
  }
  p.forgot-password-link {
    cursor: pointer;
  }
  .mb-2{
    margin-bottom: 0.5rem;
  }
}
@media only screen and (min-width: 601px) {
  #login-modal{
    width:360px;
  }
}
@media only screen and (min-width: 993px) {
  #login-modal{
    width:380px;
  }
}
