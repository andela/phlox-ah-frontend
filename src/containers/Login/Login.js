import React, { Component } from 'react';
import {
  Modal, Row, Button, Col, Input
} from 'react-materialize';
import './Login.scss';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { login } from './BasePath';
/**
 *
 *
 * @class Login
 * @extends {Component}
 */
class Login extends Component {
  /**
   * @member of Login
   */

  /**
   * @description - This method runs first in the class
   * @param {object} props
   * @returns {object} articles
   * @memberof Login
   */
  constructor(props) {
    super(props);

    this.state = {
      emailOrUsername: '',
      password: '',
    };

    this.onChange = this.onChange.bind(this);
  }

  /**
   * @description - This method sets the input values
   * @param {objecj} e
   * @returns {object} null
   * @memberof Login
   */
  onChange(e) {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  /**
   * @description - This method hides the login modal
   * @returns {object} null
   * @memberof Login
   */
  // eslint-disable-next-line
  onCloseClicked() {
    // eslint-disable-next-line
    $("#login-modal").modal("close");
  }

  /**
   * @description - This method makes login request
   * @param {objecj} e
   * @returns {object} null
   * @memberof Login
   */
  onSubmit(e) {
    e.preventDefault();
    this.props.login(this.state);
  }

  /**
   *
   * @description - This method renders the jsx for this component
   * @returns {jsx} - jsx
   * @memberof Login
   */
  render() {
    const { emailOrUsername, password } = this.state;

    return (
      <Modal
        id='login-modal'
        className="center-align">
        <div>
          <a className="close-modal" href="#"
            onClick={this.onCloseClicked}>
            <i className="fas fa-times fa-lg black-text"></i>
          </a>
        </div>
        <h5>Authors Haven</h5>
        <form id="test" className="col s12" onSubmit={this.onSubmit.bind(this)}>
          <Row>
            <Input
              type="text"
              label="username / email"
              s={12}
              name="emailOrUsername"
              value={emailOrUsername}
              onChange={this.onChange}
            />
            <Input
              type="password"
              label="password"
              s={12}
              name="password"
              value={password}
              onChange={this.onChange}
            />
            <Button
              type="submit"
              id="login-button" waves='light'>
              Login
              <i className="fas fa-sign-in-alt"></i>
            </Button>
          </Row>
        </form>
        <h6>Login Using</h6>
        <Row>
          <Col s={4} >
            <a href="#">
              <i className="fab fa-facebook fa-3x"></i>
            </a>
          </Col>
          <Col s={4}>
            <a href="#">
              <i className="fab fa-google-plus-square fa-3x"></i>
            </a>
          </Col>
          <Col s={4}>
            <a href="#">
              <i className="fab fa-twitter-square fa-3x"></i>
            </a>
          </Col>
        </Row>
        <h6>
          No account yet?
          <a href="#">
            <span className="theme-color">
              &nbsp; Sign Up
            </span>
          </a>
        </h6>
      </Modal>
    );
  }
}

Login.propTypes = {
  login: PropTypes.func.isRequired,
};

export default connect(null, { login })(Login);
