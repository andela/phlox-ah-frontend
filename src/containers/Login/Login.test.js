import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Login from './Login';
import { asyncActions, USER } from './BasePath';

const mockStore = configureMockStore();
const store = mockStore({});

let component;
let myComponent;

describe('<Login/>', () => {
  beforeEach(() => {
    component = shallow(
      <Provider store={store}>
        <Login />
      </Provider>
    );
    myComponent = component.dive({ context: { store } }).dive();
  });
  it('should render without throwing an error', () => {
    expect(component).toMatchSnapshot();// eslint-disable-line 
  });
  it('should have a form tag with class col', () => {
    expect(myComponent.find('form').hasClass('col')).toBe(true)// eslint-disable-line 
  });
  it('should have a login button', () => {
    expect(myComponent.find('Button#login-button').exists()).toBe(true)// eslint-disable-line 
  });
  it('should have a input field', () => {
    expect(myComponent.find('Input').exists()).toBe(true)// eslint-disable-line 
  });
  it('should create an action to set user object on the store', () => {
    const payload = {
      username: 'dimeji',
      email: 'dimeji@email.com',
    };
    const expectedAction = {
      type: 'USER_SUCCESS',
      payload
    };
    expect(asyncActions(USER).success(payload)).toEqual(expectedAction);// eslint-disable-line
  });
});
