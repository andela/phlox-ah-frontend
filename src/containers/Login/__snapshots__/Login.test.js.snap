// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> should render without throwing an error 1`] = `<Connect(Login) />`;
